src/internal/m365/collection/drive/item.go

```diff
@@ -193,3 +210,27 @@ func setName(orig models.ItemReferenceable, driveName string) models.ItemReferen
 
 	return orig
 }
+
+// isURLExpired inspects the jwt token embed in the item download url
+// and returns true if it is expired.
+func isURLExpired(
+	ctx context.Context,
+	url string,
+) (bool, error) {
+	// Extract the raw JWT string from the download url.
+	rawJWT, err := common.GetQueryParamFromURL(url, onedrive.JWTQueryParam)
+	if err != nil {
+		logger.CtxErr(ctx, err).Info("query param not found")
+
+		return false, clues.Stack(err)
```

Comment:
nit: could you please add the context's clues to the error in this return and the one below?
```suggestion
		return false, clues.Stack(err).WithClues(ctx)
```